nnef/src/ops/nnef/deser.rs

```diff
@@ -411,14 +424,16 @@ fn pool_spec_for_pools(
             before.push(p[0]);
             after.push(p[1]);
         }
-        PaddingSpec::Explicit(before, after, false)
+        let before_hw = get_hw_dims(&before)?;
+        let after_hw = get_hw_dims(&after)?;
+        PaddingSpec::Explicit(before_hw, after_hw, false)
     };
     Ok(PoolSpec::new(
         DataFormat::NCHW,
-        shape[2..].into(),
+        shape_hw,
         padding,
-        if dilation.len() > 2 { Some(dilation[2..].into()) } else { None },
-        if stride.len() > 2 { Some(stride[2..].into()) } else { None },
+        if dilation_hw.iter().all(|it| *it == 1) || dilation_hw.len() == 0 { None } else { Some(dilation_hw) },
```

Comment:
Are you sure about these ? dilations and strides in Rust are 1 in the "trivial" case, not 0.

Reply:
Yes I think ! If all dilations/strides are 1 then there is nothing to do so it can be `None`.